src/model/group.cpp

```diff
@@ -89,30 +87,37 @@ QString Group::getDisplayedName() const
     return getName();
 }
 
+/**
+ * @brief performs a peerId to ToxPk lookup
+ * @param peerId peerId to lookup
+ * @return ToxPk if peerId found
+ * @note should not be used, reference peers by their ToxPk instead
+ */
+const ToxPk Group::resolvePeerId(int peerId) const
+{
+    const Core* core = Core::getInstance();
+    return core->getGroupPeerPk(groupId, peerId);
+}
+
 void Group::regeneratePeerList()
```

Comment:
Just a note for the future: I think peer list generation should be moved out of the `Group`. IMO `Group` should provide info, but no behaviour

Reply:
I agree, but as you said, it's for the future